jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -1456,6 +1405,10 @@ ecma_op_function_construct (ecma_object_t *func_obj_p, /**< Function object */
     /* This is not a super call so update the "new.target" */
     JERRY_CONTEXT (current_new_target) = func_obj_p;
   }
+  else
+  {
+    arguments_list_p = ecma_op_function_args_set_flag (arguments_list_p, ECMA_FUNC_ARG_FLAG_SUPER);
```

Comment:
If I read this correctly then: the `JERRY_CONTEXT (current_new_target)` is set inside the `..._call_simple` method. If so, then the contents of the `if` above is not needed. In addition with this change the `current_new_target` is saved twice onto the stack (once here and once in the `..._call_simple`).

Reply:
good idea.